src/parser/HLSTree.h

```diff
@@ -30,10 +29,8 @@ class ATTR_DLL_LOCAL HLSTree : public AdaptiveTree
     ENCRYPTIONTYPE_UNKNOWN = 4,
   };
   HLSTree(const UTILS::PROPERTIES::KodiProperties& kodiProps,
-          CHOOSER::IRepresentationChooser* reprChooser,
-          IAESDecrypter* decrypter)
-    : AdaptiveTree(kodiProps, reprChooser), m_decrypter(decrypter){};
-  HLSTree(const HLSTree& left);
+          CHOOSER::IRepresentationChooser* reprChooser);
+  HLSTree(const HLSTree& left) : AdaptiveTree(left.m_kodiProps, left.m_reprChooser){};
```

Comment:
this method was on cpp, can you keep it on cpp?

Reply:
Sure thing